src/test/unit_tests/test_bucket_replication.js

```diff
@@ -798,29 +798,122 @@ async function put_object(s3_owner, bucket_name, key, optional_body) {
+  }
+
 async function list_objects_and_wait(s3_owner, bucket, expected_num_of_objects) {
     let res;
     for (let retries = 1; retries <= 3; retries++) {
         try {
-            res = await list_objects(s3_owner, bucket);
+            res = await list_all_objs_in_bucket(s3_owner, bucket);
             console.log('list_objects_and_wait: ', res);
-            assert.deepStrictEqual(res.Contents.length, expected_num_of_objects);
-            console.log(`list_objects contents: expected: ${expected_num_of_objects} actual: ${res.Contents.length} ${res.Contents}`);
-            return res.Contents;
+            assert.deepStrictEqual(res.length, expected_num_of_objects);
+            console.log(`list_objects contents: expected: ${expected_num_of_objects} actual: ${res.length} ${res}`);
+            return res;
         } catch (e) {
-            console.log(`waiting for replications of bucket: ${bucket}, response: ${util.inspect(res)} `); //num of retries: ${retries}`);
+            console.log(`waiting for replications of bucket: ${bucket}, response: ${res} `); //num of retries: ${retries}`);
             if (retries === 3) throw e;
             await P.delay(2 * 1000);
         }
     }
 }
+
+mocha.describe('replication 1.1k object test', function() {
+    const self = this; // eslint-disable-line no-invalid-this
+    self.timeout(180000);
+    const obj_amount = 1100;
+    const src_bucket = 'src-bucket';
+    const target_bucket = 'target-bucket';
+    const buckets = [src_bucket, target_bucket];
+    let s3_owner;
+    let scanner;
+    let s3_creds = {
+        s3ForcePathStyle: true,
+        signatureVersion: 'v4',
+        computeChecksums: true,
+        s3DisableBodySigning: false,
+        region: 'us-east-1',
+        httpOptions: { agent: new http.Agent({ keepAlive: false }) },
+    };
+    let src_bucket_keys = [];
+    let target_bucket_keys = [];
+    mocha.before('init scanner & populate buckets', async function() {
+        // create buckets
+        await P.all(_.map(buckets, async bucket_name => {
+            await rpc_client.bucket.create_bucket({ name: bucket_name });
+        }));
+        await put_replication(src_bucket, [
+            { rule_id: '1k-replication-rule', destination_bucket: target_bucket, filter: { prefix: '' } }
+        ], false);
+        const admin_account = await rpc_client.account.read_account({ email: EMAIL });
+        const admin_keys = admin_account.access_keys;
+        //await create_namespace_buckets(admin_account);
+
+        s3_creds.accessKeyId = admin_keys[0].access_key.unwrap();
+        s3_creds.secretAccessKey = admin_keys[0].secret_key.unwrap();
+        s3_creds.endpoint = coretest.get_http_address();
+        s3_owner = new AWS.S3(s3_creds);
+
+        // populate source bucket
+        for (let i = 0; i < obj_amount; i++) {
+            let key = create_random_body();
+            src_bucket_keys.push(key);
+            await put_object(s3_owner, src_bucket, key);
```

Comment:
How long does it take to run it with 1100 objects? Can we do it by using a smaller number of objects and changing `process.env.REPLICATION_MAX_KEYS`?